### PR TITLE
RND-650 Add DDS Badge component

### PR DESCRIPTION
**Pull request by qooban**
## Description

Created `Badge` component based on [DDS Badge component](https://www.delldesignsystem.com/components/badge).

## Screenshots / Videos
image

## Checklist
- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
HTML snapshots and Loki reference images added.

## Documentation
JSDocs for Badge props.

### CODE REVIEW

File: src/components/dds/Badge/Badge.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    /**
+     * Size of the badge
+     */
+    size?: Size;
+
+    /**
+     * Content of the badge
+     */
+    emphasis?: Emphasis;
```
[the remark this answers is not in the document]
Comment (qooban) [reply src/components/dds/Badge/Badge.tsx]: No, fixed it.

File: src/components/dds/Badge/Badge.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    return (
+        <span
+            className={getRootClassNames(
+                componentClasses.base,
+                componentClasses.sizes[size],
+                color && componentClasses.colors[color],
+                emphasis && componentClasses.emphasis[emphasis],
+                marginClasses,
+                className
+            )}
```
[the remark this answers is not in the document]
Comment (qooban) [reply src/components/dds/Badge/Badge.tsx]: Sure, changed.

File: src/components/dds/Badge/Badge.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            {...rest}
+        >
+            {iconName && <Icon name={iconName} className={componentClasses.badgeIcon} />}
```
[the remark this answers is not in the document]
Comment (qooban) [reply src/components/dds/Badge/Badge.tsx]: I'll add support for it.

File: src/components/dds/Badge/Badge.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+import type { SpacingSize } from '../common/spacing';
+import { getMarginClassName } from '../common/spacing';
+
+// NOTE: Duplicating types from 'dds-core/components/badge/ts/badge' as const definition lacks "as const"
+export const EmphasisValues = ['heavy', 'medium', 'light'] as const;
+export const ColorValues = ['informative', 'success', 'warning', 'error', 'gray', 'brand'] as const;
+export const SizeValues = ['microdot', 'dot', 'mini', 'sm', 'md', 'lg'] as const;
+export type Emphasis = typeof EmphasisValues[number];
+export type Color = typeof ColorValues[number];
+export type Size = typeof SizeValues[number];
```
[the remark this answers is not in the document]
Comment (qooban) [reply src/components/dds/Badge/Badge.tsx]: Just copied it from DDS core, but yeah, let's define it properly and extend the comment.

File: src/components/dds/Badge/Badge.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+    /**
+     * Name of an icon to be displayed in the button
+     */
+    iconName?: Icons;
+
+    /**
+     * Type of an icon to be displayed in the button
+     */
+    iconType?: IconType;
```
Comment (kubama) [inline src/components/dds/Badge/Badge.tsx:55]: Please note my comment in another PR: https://github.com/cloudify-cosmo/cloudify-ui-components/pull/219#discussion_r1194880642

**Review by kubama — APPROVED** [review]

File: src/components/dds/Badge/Badge.tsx (hunk 5 above)
Comment (qooban) [reply src/components/dds/Badge/Badge.tsx:55]: Good idea. Addressed it here.

FYI @PatrycjaKubiczek

**Review by kubama — DISMISSED** [review]

**Review by Vorbert-Kruk — COMMENTED** [review]
Will do review tomorrow

**Comment by PatrycjaKubiczek** [comment]
will review tomorrow as well

**Review by PatrycjaKubiczek — APPROVED** [review]
LGTM 👍

**Review by Vorbert-Kruk — APPROVED** [review]

**Pull request merged**